src/gotcha_auxv.c

```diff
@@ -227,7 +227,7 @@ struct link_map *get_vdso_from_maps()
    struct link_map *m;
    maps = gotcha_open("/proc/self/maps", O_RDONLY);
    for (;;) {
-      hit_eof = read_line(line, 4097, maps);
+      hit_eof = read_line(line, 4096, maps);
```

Comment:
This looks fine to me. Just that, to avoid this kind of bug, it might be better to define a constant?

Reply:
The read_line function writes a zero-byte to size-1, i.e. line[4096] which is out-of-bounds. So it is a bug.

Reply:
Oh i meant, your fix looks good. I was saying, we should change the line array as well to use a constant.

```
#define BUFFER_LEN 4096
char name[BUFFER_LEN], line[BUFFER_LEN], *line_pos;
   struct link_map *m;
   maps = gotcha_open("/proc/self/maps", O_RDONLY);
   for (;;) {
      hit_eof = read_line(line, BUFFER_LEN, maps);
```

As I see this buffer length used in more places in the file like ```parse_auxv_contents()```

Reply:
Oh, OK. Yeah I guess it makes sense to use a constant.

Reply:
I did the fix.